src/de/mangatube/src/eu/kanade/tachiyomi/extension/de/mangatube/MangaTube.kt

```diff
@@ -40,146 +35,225 @@ class MangaTube : ParsedHttpSource() {
-        return POST("$baseUrl/ajax", xhrHeaders, rbodyContent.toRequestBody(null))
-    }
+        val url = "$baseUrl/api/home/top-manga"
 
-    // popular uses "success" as a key, search uses "suggestions"
-    // for future reference: if adding filters, advanced search might use a different key
-    private fun parseMangaFromJson(response: Response, hasNextPage: Boolean): MangasPage {
-        var titleKey = "manga_title"
-        val mangas = json.decodeFromString<JsonObject>(response.body.string())
-            .let { it["success"] ?: it["suggestions"].also { titleKey = "value" } }!!
-            .jsonArray
-            .map { json ->
-                SManga.create().apply {
-                    title = json.jsonObject[titleKey]!!.jsonPrimitive.content
-                    url = "/series/${json.jsonObject["manga_slug"]!!.jsonPrimitive.content}"
-                    thumbnail_url = json.jsonObject["covers"]!!.jsonArray[0].jsonObject["img_name"]!!.jsonPrimitive.content
-                }
-            }
-        return MangasPage(mangas, hasNextPage)
+        Log.d("MangaTube", "Request popular mangas")
+        Log.d("MangaTube", "Url -> $url")
+
+        return GET(url)
     }
 
-    override fun popularMangaSelector() = throw UnsupportedOperationException()
+    override fun popularMangaParse(response: Response): MangasPage {
+        val body = MangaTubeHelper.checkResponse(response)
 
-    override fun popularMangaFromElement(element: Element): SManga = throw UnsupportedOperationException()
+        val res: BaseResponse<MangasWrapper> = json.decodeFromString(body)
 
-    override fun popularMangaNextPageSelector() = throw UnsupportedOperationException()
+        if(!res.success){
+            throw Exception("Something went wrong!")
+        }
+
+
+        val mangaList = res.data.manga.map { manga ->
+            mangas[manga.title] = manga
+            SManga.create().apply {
+                title = manga.title
+                url = "$baseUrl${manga.url}"
+                thumbnail_url = manga.cover
+                genre = manga.genre.map { genre -> Genre.fromId(genre)!! }
+                    .joinToString(", ") { genre -> genre.displayName }
+                status = MangaTubeHelper.mangaStatus(manga.status)
+            }
+        }
 
-    // Latest
+        return MangasPage(mangaList, false)
+    }
 
     override fun latestUpdatesRequest(page: Int): Request {
-        return GET("$baseUrl/?page=$page", headers)
+        val url = "$baseUrl/api/home/new-manga"
+
+        Log.d("MangaTube", "Request new mangas")
+        Log.d("MangaTube", "Url -> $url")
+
+        return GET(url)
     }
 
-    override fun latestUpdatesSelector() = "div#series-updates div.series-update:not([style\$=none])"
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val body = MangaTubeHelper.checkResponse(response)
```

Comment:
consider calling this method in an interceptor since you are checking response every time